app/models/request.rb

```diff
@@ -11,12 +11,15 @@
 #  receiver_id :bigint           not null
 #
 class Request < ApplicationRecord
+# Associations
   belongs_to :item
   belongs_to :giver, class_name: "User"
   belongs_to :receiver, class_name: "User"
-  validates_presence_of :giver_id, :item_id, :receiver_id
   # one request can have two feedbacks (from receiver and giver)
-  has_many :feedbacks
+  has_many :feedbacks, dependent: :destroy
   # one chatroom represents one chatroom and has many messages
   has_many :messages
```

Comment:
@DenzelBraithwaite 
`has_many :feedbacks` to remove `dependent: :destroy` because we want to keep the feedback instances in our DB for generating badges.
`has_many :messages`: please add `dependent: :destroy` as discussed.